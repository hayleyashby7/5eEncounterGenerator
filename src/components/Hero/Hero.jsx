import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from '../Form/Button';

function Hero({ clickCB }) {
    const [isMinimised, setIsMinimised] = useState(false);

    const heroClick = () => {
        setIsMinimised(true);
        clickCB();
    };

    return (
        <section
            className={`${
                isMinimised ? 'h-[20vh] ' : 'h-screen'
                // eslint-disable-next-line prettier/prettier
            }  bg-dragon bg-cover bg-right bg-no-repeat`}
        >
            <div
                className={`${
                    isMinimised ? 'h-[20vh]' : 'h-screen '
                }  flex flex-col bg-neutral-900/50`}
            >
                <header className=' flex flex-none flex-col text-center'>
                    <h1 className="font-['Goblin_One'] text-4xl lg:text-8xl">DDEG</h1>
                    <h2 className="font-['Alegreya_Sans_SC'] text-2xl">
                        Dungeons & Dragons Encounter Generator
                    </h2>
                </header>

                {!isMinimised && (
                    <div className='flex flex-1 grow flex-col justify-evenly text-center'>
                        <h2 className="font-['Alegreya_Sans_SC'] text-2xl lg:text-5xl">
                            Take the stress out of being a DM
                        </h2>
                        <p className=" font-['Alegreya_Sans'] text-xl lg:text-3xl">
                            Generate an encounter customised to YOUR party
                        </p>
                        <span>
                            <Button
                                type='button'
                                label='Build Your Encounter'
                                onClick={heroClick}
                            />
                        </span>
                    </div>
                )}
            </div>
        </section>
    );
}

Hero.propTypes = {
    clickCB: PropTypes.func.isRequired,
};

export default Hero;
