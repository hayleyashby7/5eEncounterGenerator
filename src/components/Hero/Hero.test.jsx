import { describe, it, expect, vi, beforeEach } from 'vitest';
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Hero from './Hero';

describe('Hero component', () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });
    it('renders the component', () => {
        // Arrange
        const mockClickCB = vi.fn();
        const { container } = render(<Hero clickCB={mockClickCB} />);

        // Assert
        expect(container).toMatchSnapshot();
    });

    it('should call the clickCB function when the button is clicked', async () => {
        // Arrange
        const user = userEvent.setup();
        const mockClickCB = vi.fn();

        // Act
        render(<Hero clickCB={mockClickCB} />);

        await user.click(screen.getByRole('button', { name: 'Build Your Encounter' }));

        // Assert
        expect(mockClickCB).toBeCalled();
        expect(mockClickCB).toHaveBeenCalledTimes(1);
    });

    it('should minimise the hero component when the button is clicked', async () => {
        // Arrange
        const user = userEvent.setup();
        const mockClickCB = vi.fn();

        // Act
        render(<Hero clickCB={mockClickCB} />);

        await user.click(screen.getByRole('button', { name: 'Build Your Encounter' }));

        // Assert
        expect(
            screen.queryByRole('button', { name: 'Build Your Encounter' }),
        ).not.toBeInTheDocument();
        expect(screen.queryByText(/Take the stress out of being a DM/i)).not.toBeInTheDocument();
        expect(
            screen.queryByText(/Generate an encounter customised to YOUR party/i),
        ).not.toBeInTheDocument();
    });
});
