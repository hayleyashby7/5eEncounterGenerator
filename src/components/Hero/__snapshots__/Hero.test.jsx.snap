// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero component > renders the component 1`] = `
<div>
  <section
    class="h-screen  bg-dragon bg-cover bg-right bg-no-repeat"
  >
    <div
      class="h-screen   flex flex-col bg-neutral-900/50"
    >
      <header
        class=" flex flex-none flex-col text-center"
      >
        <h1
          class="font-['Goblin_One'] text-4xl lg:text-8xl"
        >
          DDEG
        </h1>
        <h2
          class="font-['Alegreya_Sans_SC'] text-2xl"
        >
          Dungeons & Dragons Encounter Generator
        </h2>
      </header>
      <div
        class="flex flex-1 grow flex-col justify-evenly text-center"
      >
        <h2
          class="font-['Alegreya_Sans_SC'] text-2xl lg:text-5xl"
        >
          Take the stress out of being a DM
        </h2>
        <p
          class=" font-['Alegreya_Sans'] text-xl lg:text-3xl"
        >
          Generate an encounter customised to YOUR party
        </p>
        <span>
          <button
            class="rounded-lg border-2 border-solid border-orange-100 bg-red-950 p-2 font-['Alegreya_Sans'] text-orange-100 hover:bg-red-900 hover:text-orange-200 focus:outline-none focus:ring-2 focus:ring-red-900/50"
            rest="[object Object]"
            type="button"
          >
            Build Your Encounter
          </button>
        </span>
      </div>
    </div>
  </section>
</div>
`;
